Cpp_test/lib/ipc-queue-receive.cpp

```diff
@@ -12,49 +12,57 @@ void ipcQueueReceive::init()
 {
 	this->file_handler.setup_file(info.file_name, FileMode::WRITE);
 	std::cout << "Waiting for sender" << std::endl;
+	int time_wait = 0;
 	do
 	{
 		this->mqd = mq_open(this->info.method_name, O_RDONLY, 0660, &(this->attr));
-		int i = 0;
-		i++;
-		if (i == 10)
-		{
-			throw std::runtime_error("ERROR: Fail to open receive queue.");
-		}
-	} while (this->mqd == -1);
 
-	if (this->mqd == -1){
+		sleep(1);
+		time_wait++;
+
+		std::cout << time_wait << std::endl;
+	} while (this->mqd == -1 && time_wait < 10);
+
+	if (time_wait == 10)
```

Comment:
The same way that in pipe, for safety reasons it is better to also include values bigger than 10.
```suggestion
	if (time_wait >= 10)
```